pkg/controller/build/build_controller.go

```diff
@@ -218,3 +244,91 @@ func (r *ReconcileBuild) validateOutputSecret(n string, ns string) error {
 	}
 	return nil
 }
+
+func (r *ReconcileBuild) configFinalizer(b *build.Build) error {
+	if b.Annotations != nil && b.Annotations[build.AnnotationBuildRunDeletion] == "true" {
```

Comment:
why you dont get annotations via `b.ObjectMeta.Annotations`? Also, can `annotations` be nil?

Reply:
`b.ObjectMeta.Annotations` and `b.Annotations` are same, you can see from the IDE.

annotations can be nil, if there is no any item in the annotations, or no `annotations` setting in the resource, it will be the `nil`

Reply:
ok, I was not aware it accessed also without going to ObjectMeta. About the `nil`, I think you need to assert like `b.Annotations != nil` , is not needed. See an example https://play.golang.org/p/bmywWEQSdwC

Reply:
Yes, I also checked, we don't need to check it is nil, changed to:
`if b.GetAnnotations()[build.AnnotationBuildRunDeletion] == "true"`
Thanks!